### PR TITLE
#1959 enhanced numeric option setting

### PR DESCRIPTION
**Pull request by crazycatdevs**
#1959 enhanced numeric option setting

### CODE REVIEW

File: admin/modules/config/settings.php [hunk 1]
```diff
@@ -1182,7 +1182,20 @@
...
+				$field_options = array('id' => $element_id);
+                            if (count($type)>1)
+                            {
+                                for($i=1; $i < count($type); $i++)
+                                {
+                                    $optionsexp = explode("=", $type[$i]);
+                                    $opt = trim($optionsexp);
+                                    if (in_array($opt[0], array('min', 'max', 'step')))
+                                    {
+                                        $field_options[$opt[0]] = intval($opt[1]);
```
Comment (ATofighi) [inline admin/modules/config/settings.php]: it should be:
```php
                                        $field_options[$opt[0]] = (int)$opt[1];
```
`(int)` is faster than `intval()`.

Comment (crazycatdevs) [inline admin/modules/config/settings.php]: Done. I'll try to keep that in mind

File: admin/modules/config/settings.php [hunk 2]
```diff
@@ -1182,7 +1182,20 @@
...
+				$field_options = array('id' => $element_id);
+                            if (count($type)>1)
+                            {
+                                for($i=1; $i < count($type); $i++)
+                                {
+                                    $optionsexp = explode("=", $type[$i]);
+                                    $opt = trim($optionsexp);
+                                    if (in_array($opt[0], array('min', 'max', 'step')))
+                                    {
+                                        $field_options[$opt[0]] = int()$opt[1];
```
Comment (ATofighi) [inline admin/modules/config/settings.php]: `(int)$opt[1]` :!

**Comment by crazycatdevs** [comment]
Sorry, it was a long day. Corrected.

**Comment by ATofighi** [comment]
please use `tabs` instead of `spaces`....
http://docs.mybb.com/1.8/development/standards/#indenting

**Comment by Destroy666x** [comment]
Note that existing settings could (should?) be update to use these new options, especially `min`

**Comment by crazycatdevs** [comment]
@Destroy666x : I'll try to list settings which could have these new options, but it may take time ;)

**Comment by Destroy666x** [comment]
> edit : the numeric fields in the admin settings already use the min (default: 0). Only the plugins are affected

Haven't noticed your edit, but no option uses `min` because it hasn't been implemented yet. You probably see `0`s because of a bug which is going to be fixed in another PR (all empty values are casted to 0 even though they shouldn't be).

**Comment by JN-Jones** [comment]
@crazycatdevs

**Comment by JN-Jones** [comment]
@crazycatdevs Can you update this PR with the comments made on the issue?

**Comment by JN-Jones** [comment]
New PR for this: #2193

**Pull request closed without merging**